ui/src/data-services/models/taxa.ts

```diff
@@ -13,26 +27,26 @@ export class Taxon {
   readonly ranks: { id: string; name: string; rank: string }[]
 
   public constructor(taxon: ServerTaxon) {
-    this.id = `${taxon.id}`
+    this.id = taxon.id
     this.name = taxon.name
     this.parentId = taxon.parent ? `${taxon.parent?.id}` : undefined
     this.rank = taxon.rank
-    this.ranks = this._getRanks(taxon)
-  }
-
-  private _getRanks(taxon: ServerTaxon) {
-    const ranks = []
 
-    let current: ServerTaxon | undefined = taxon.parent
-    while (current) {
-      ranks.push({
-        id: `${current.id}`,
-        name: current.name,
-        rank: current.rank,
-      })
-      current = current.parent
+    if (taxon.parents) {
+      this.ranks = taxon.parents
+    } else if (taxon.parent?.parents) {
+      // TODO: Update this when species list is returning parents similar to other endpoints
```

Comment:
@mihow seems in the species list case, data structure looked slightly different. I could still access all parents though from the `parent` object. We can skip this once we streamline the format :)